monai/apps/pathology/transforms/post/array.py

```diff
@@ -625,77 +626,182 @@ def __call__(  # type: ignore
+            activation=activation, threshold=mask_threshold, min_object_size=min_object_size
+        )
+        self.generate_instance_border = GenerateInstanceBorder(kernel_size=sobel_kernel_size)
+        self.generate_distance_map = GenerateDistanceMap(smooth_fn=distance_smooth_fn)
+        self.generate_watershed_markers = GenerateWatershedMarkers(
+            threshold=marker_threshold,
+            radius=marker_radius,
+            postprocess_fn=marker_postprocess_fn,
+            min_object_size=min_object_size,
+        )
+        self.watershed = Watershed(connectivity=watershed_connectivity)
+        self.generate_instance_contour = GenerateInstanceContour(
+            min_num_points=min_num_points, contour_level=contour_level
+        )
         self.generate_instance_centroid = GenerateInstanceCentroid()
+
+    def __call__(  # type: ignore
+        self, nuclear_prediction: NdarrayOrTensor, hover_map: NdarrayOrTensor
+    ) -> Tuple[Dict, NdarrayOrTensor]:
+        """post-process instance segmentation branches (NP and HV) to generate instance segmentation map.
+
+        Args:
+            nuclear_prediction: the output of NP (nuclear prediction) branch of HoVerNet model
+            hover_map: the output of HV (hover map) branch of HoVerNet model
+        """
+
+        # Process NP and HV branch using watershed algorithm
+        watershed_mask = self.generate_watershed_mask(nuclear_prediction)
+        instance_borders = self.generate_instance_border(watershed_mask, hover_map)
+        distance_map = self.generate_distance_map(watershed_mask, instance_borders)
+        watershed_markers = self.generate_watershed_markers(watershed_mask, instance_borders)
+        instance_map = self.watershed(distance_map, watershed_mask, watershed_markers)
+
+        # Create bounding boxes, contours and centroids
+        instance_ids = set(np.unique(instance_map)) - {0}  # exclude background
+        instance_info = {}
+        for inst_id in instance_ids:
+            instance_mask = instance_map == inst_id
+            instance_bbox = BoundingRect()(instance_mask)
+
+            instance_mask = instance_mask[
+                :, instance_bbox[0][0] : instance_bbox[0][1], instance_bbox[0][2] : instance_bbox[0][3]
+            ]
+            offset = [instance_bbox[0][2], instance_bbox[0][0]]
+            instance_contour = self.generate_instance_contour(instance_mask, offset)
+            if instance_contour is not None:
+                instance_centroid = self.generate_instance_centroid(instance_mask, offset)
+                instance_info[inst_id] = {
+                    "bounding_box": instance_bbox,
+                    "centroid": instance_centroid,
+                    "contour": instance_contour,
+                }
+
+        return instance_info, instance_map
+
+
+class HoVerNetTypeMapPostProcessing(Transform):
+    """
+    The post-processing transform for HoVerNet model to generate a segmentation map with types.
```

Comment:
I'm a little confused by "a segmentation map with types", perhaps “a pixel-level type map” could be used?